### PR TITLE
Re-add netframework targeting for Tools.proj PackageRef compat

### PR DESCRIPTION
**Pull request by mmitche**
### To double check:

* [ ] The right tests are in and the right validation has happened.  Guidance: https://github.com/dotnet/arcade/blob/main/Documentation/Validation.md

### CODE REVIEW

File: src/Microsoft.DotNet.MacOsPkg/Microsoft.DotNet.MacOsPkg.csproj [hunk 1]
```diff
@@ -1,15 +1,19 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>$(NetToolCurrent)</TargetFramework>
+    <!-- We include both netframework and netcore in the target frameworks because Arcade's Tools.proj
+         'targets' via net472. -->
+    <TargetFramework>$(NetToolCurrent);$(NetFrameworkToolCurrent)</TargetFramework>
```
Comment (ellahathaway) [inline src/Microsoft.DotNet.MacOsPkg/Microsoft.DotNet.MacOsPkg.csproj:6]:
```suggestion
    <TargetFrameworks>$(NetToolCurrent);$(NetFrameworkToolCurrent)</TargetFrameworks>
```

Comment (mmitche) [reply src/Microsoft.DotNet.MacOsPkg/Microsoft.DotNet.MacOsPkg.csproj:6]: agh good catch

File: src/Microsoft.DotNet.MacOsPkg/Microsoft.DotNet.MacOsPkg.csproj [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
     <Description>The MacOsPkg tool is used for unpacking and packing MacOS .pkg files and nested .app bundles.</Description>
     <PackageTags>Arcade Build Tool MacOS Pkg</PackageTags>
     <DevelopmentDependency>false</DevelopmentDependency>
-    <PackAsTool>true</PackAsTool>
-    <ToolCommandName>dotnet-macos-pkg</ToolCommandName>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(TargetFramework)' == '$(NetToolCurrent)'">
+    <PackAsTool>true</PackAsTool>
+    <ToolCommandName>dotnet-tar</ToolCommandName>
```
Comment (ellahathaway) [inline src/Microsoft.DotNet.MacOsPkg/Microsoft.DotNet.MacOsPkg.csproj:17]:
```suggestion
    <ToolCommandName>dotnet-macos-pkg</ToolCommandName>
```

**Review by ellahathaway — COMMENTED** [review]
Last one then LGTM

**Review by ellahathaway — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Isn't it possible to target a .NETCoreApp TFM in Tools.proj instead?

**Pull request merged**